cli/command/cli.go

```diff
@@ -549,7 +549,9 @@ func resolveContextName(opts *cliflags.CommonOptions, config *configfile.ConfigF
 	if config != nil && config.CurrentContext != "" {
 		_, err := contextstore.GetMetadata(config.CurrentContext)
 		if store.IsErrContextDoesNotExist(err) {
-			return "", errors.Errorf("Current context %q is not found on the file system, please check your config file at %s", config.CurrentContext, config.Filename)
+			// If config file references a context that does not exist, fallback to default
+			// Explicitly requested contexts (flag or env-var based) will still get the error
+			return DefaultContextName, nil
```

Comment:
I don't think we should silently fall back, as this would lead to unexpected results (similar to outlined in https://github.com/docker/cli/issues/1637)

Reply:
Would writing a message to stderr feel good enough ?

Reply:
No, I don't think a message would be enough, because that wouldn't make the command fail.

Consider a situation where I am in a specific context, and that context is (somehow) removed, and all of a sudden my command completes successfully but against the wrong server. Depending on the command I ran that could mean I just (e.g.)  _removed_ a volume or image from the wrong environment.

Reply:
Indeed. Would deferring the error to the http call be enough?

Reply:
If we can defer to the moment it's actually needed, yes, that'd work (probably also when doing `docker context inspect`, but I guess that would already be the case)